netaddr.go

```diff
@@ -653,19 +653,31 @@ func appendDecimal(b []byte, x uint8) []byte {
 	return append(b, digits[x%10])
 }
 
-// appendHex appends the hex string representation of x to b.
-func appendHex(b []byte, x uint16) []byte {
+// Descriptive constants to use for the pad parameter of appendHex.
+const (
+	padHex   = true
+	noPadHex = false
+)
+
+// appendHex appends the hex string representation of x to b and optionally
+// applies leading zero padding.
+func appendHex(b []byte, x uint16, pad bool) []byte {
 	// Using this function rather than strconv.AppendUint makes IPv6
 	// string building 2x faster.
-
 	if x >= 0x1000 {
```

Comment:
It looks to me like 

```go
if x >= 0x1000 {
 	b = append(b, digits[x>>12])
 } else if pad {
 	b = append(b, '0')
}
```

is equivalent to:

```go
if x >= 0x1000 || pad {
 	b = append(b, digits[x>>12])
}
```

because `digits[0]` is '0'.

And if you rewrite this whole function that way, it starts to look like we may as well have a separate padded version from unpadded version. The padded version is a single call to append with four digits lookups in it. And then we don't have a bool param to kick around.